crates/claims/crates/status/src/client/mod.rs

```diff
@@ -0,0 +1,78 @@
+use crate::EncodedStatusMap;
+use std::{borrow::Borrow, ops::Deref, sync::Arc};
+
+mod cache;
+pub use cache::Cached;
+
+mod http;
+pub use http::HttpClient;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ProviderError {
+    #[error("not found")]
+    NotFound,
+
+    #[error("internal error: {0}")]
+    Internal(String),
+
+    #[error("missing media type")]
+    MissingMediaType,
+
+    #[error("invalid media type")]
+    InvalidMediaType,
+
+    #[error("{0}")]
+    Encoded(String),
+
+    #[error("decoding failed: {0}")]
+    Decoding(String),
+}
+
+pub trait TypedStatusMapProvider<I: ?Sized, T: EncodedStatusMap> {
+    #[allow(async_fn_in_trait)]
+    async fn get_typed(&self, id: &I) -> Result<MaybeCached<T::Decoded>, ProviderError>;
```

Comment:
I'm not familiar with the specs, why does `ssi` considers caching in the core trait instead of letting the implementer bother with it?

Reply:
Caching is part of the spec yes. The trait itself doesn't provide caching by itself, but the `MaybeCached` type allows it without need to clone anything. Maybe a type parameter would be better but I try to avoid them.